addons/nametags/functions/fnc_initIsSpeaking.sqf

```diff
@@ -31,46 +31,39 @@ if (!hasInterface) exitWith {};
 ["playerChanged", {
     //When player changes, make sure to reset old unit's variable
     PARAMS_2(_newUnit,_oldUnit);
-    if (_oldUnit getVariable [QGVAR(isSpeaking), false]) then {
-        _oldUnit setVariable [QGVAR(isSpeaking), false, true];
+    if ((!isNull _oldUnit) && {_oldUnit getVariable [QGVAR(isSpeakingInGame), false]}) then {
+        _oldUnit setVariable [QGVAR(isSpeakingInGame), false, true];
     };
 }] call EFUNC(common,addEventHandler);
 
+//PFEH to watch the internal VON icon
+//Note:  class RscDisplayVoiceChat {idd = 55};  //only present when talking
+[{
+    _oldSetting = ACE_player getVariable [QGVAR(isSpeakingInGame), false];
+    _newSetting = (!(isNull findDisplay 55));
+    if (!(_oldSetting isEqualTo _newSetting)) then {
+        ACE_player setVariable [QGVAR(isSpeakingInGame), _newSetting, true];
+    };
+} , 0.1, []] call CBA_fnc_addPerFrameHandler;
```

Comment:
Although not 100% reliable, I'd not spawn this PFH if either TFAR or ACRE are present. AFAIK the chance somebody would use normal VON in that situation is very low.